streams/shared/src/main/scala/zio/stream/ZSink.scala

```diff
@@ -1835,6 +1848,20 @@ object ZSink extends ZSinkPlatformSpecificConstructors with Serializable {
       def cont(state: State)           = false
     }
 
+  /**
+   * Creates a sink which sums elements, provided they are Numeric
+   */
+  def sum[A](implicit ev: Numeric[A]): ZSink[Any, Nothing, Nothing, A, A] = {
```

Comment:
To cut down on repetition, let's implement this with `Sink.foldLeft`?

Reply:
👍